HW5/HW5.sql

```diff
@@ -0,0 +1,113 @@
+--startup:
+--mysql.server start
+
+USE baseball;
+
+DROP TABLE IF EXISTS rolling_25_prior_days_games;
+
+CREATE TABLE rolling_25_prior_days_games as
+SELECT a.game_id, b.game_id as prior_game, a.local_date, b.local_date as prior_game_local_date
+from game a,
+     game b
+where a.local_date > b.local_date
+  and b.local_date > date_add(a.local_date, INTERVAL -25 DAY);
+
+DROP INDEX IF EXISTS pg1 on rolling_25_prior_days_games;
+
+CREATE INDEX pg1 on rolling_25_prior_days_games (game_id);
+
+CREATE TABLE pitcher_counts_25_day_rolling as
+select a.game_id,
+       a.pitcher,
+       sum(b.atBat)         atBats,
+       sum(b.hit)           hits,
+       sum(b.home_run)      home_run,
+       sum(b.Intent_Walk)   Intent_Walk,
+       sum(b.Field_Error)   Field_Error,
+       sum(b.`Double`)      'double',
+       sum(b.Double_Play)   Double_Play,
+       sum(b.Single)        single,
+       sum(b.pitchesThrown) pitchesThrown,
+       sum(b.outsPlayed)    outsPlayed
+from pitcher_counts a,
+     pitcher_counts b
+where b.game_id in (select prior_game from rolling_25_prior_days_games where game_id = a.game_id)
+  and a.pitcher = b.pitcher
+group by a.game_id,
+         a.pitcher;
+
+CREATE INDEX pc25 on pitcher_counts_25_day_rolling (game_id);
+
+DROP VIEW IF EXISTS home_team_wins;
+
+CREATE VIEW home_team_wins as
+SELECT g.game_id,
+       year(g.local_date)                                                               year,
+       home.pitcher                                                                     home_starting_pitcher,
+       away.pitcher                                                                     away_starting_pitcher,
+       CASE WHEN b.winner_home_or_away = "H" then 1 else 0 END                       as HOME_TEAM_WINS,                   #response
```

Comment:
I like the case logic here, I couldn't figure out how to get it less than 3 lines